sdk/trace/span.go

```diff
@@ -255,14 +255,14 @@ func (s *span) End(options ...trace.SpanOption) {
 	}
 }
 
-// RecordError will record err as a span event for this span. If this span is
-// not being recorded or err is nil than this method does nothing.
+// RecordError will record err as a span event for this span. This will
+// not set error status to span. If this span is not being recorded or
```

Comment:
```suggestion
// not set the span status to error. If this span is not being recorded or
```

That said, I wonder whether it is worth mentioning what the method *doesn't* do at all.  It seems like it might be confusing if someone were to scan that quickly and accidentally elide the 'not'.

Reply:
I agree with you but I think @MrAlias  want to make it clear that it won't set status but just records error event.

Reply:
I could be convinced I just have a recency bias here given we are changing behavior.

Alternatively, we could potentially switch to positive phrasing here. E.g

```
An additional call to SetStatus is required if the Status of the Span should be set to Error, this method does not change the Span status.
```

@Aneurysm9 thoughts?

Reply:
> An additional call to SetStatus is required if the Status of the Span should be set to Error, this method does not change the Span status.

I'm on board with this.